js/src/ui/Form/TypedInput/typedInput.js

```diff
@@ -256,20 +269,22 @@ export default class TypedInput extends Component {
    * @see https://github.com/facebook/react/issues/1549
    */
   renderFloat (value = this.props.value, onChange = this.onChange) {
-    const { label, error, hint, min, max } = this.props;
+    const { allowCopy, label, error, hint, min, max, readOnly } = this.props;
     const param = this.getParam();
 
     const realValue = value && typeof value.toNumber === 'function'
-      ? value.toNumber()
+      ? (readOnly ? value.toFormat() : value.toNumber())
```

Comment:
I'd rather check if it's a `BigNumber` here, because you use both `toFormat` & `toNumber`.